### PR TITLE
Fix router and functional tests for AB#14979

### PR DESCRIPTION
**Pull request by BrianMaki**
# Implements [AB#14979](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/14979)

## Description

- Removed commented out code from router
- Removed bug for vaccinecard route
- Fixed null issues in before guard in router
- Fixed functional tests relating to AB314885
- Fixed functional tests for waitlist

## Testing

- [ ] Unit Tests Updated
- [x] Functional Tests Updated
- [ ] Not Required

## UI Changes

None

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Testing/functional/tests/cypress/integration/e2e/covid19/authenticatedVaccineCard.js [hunk 1]
```diff
@@ -7,7 +7,17 @@ describe("Authenticated Vaccine Card", () => {
...
+        cy.configureSettings(
+            {
+                covid19: {
+                    proofOfVaccination: {
+                        exportPdf: true,
+                    },
+                },
+            },
+            ["Immunization"]
+        );
```
Comment (MikeLyttle) [inline Testing/functional/tests/cypress/integration/e2e/covid19/authenticatedVaccineCard.js:19]: I don't think exportPdf needs to be set in this test or many of the others, only in ones that used to enable "VaccinationExportPdf".

**Review by rayking29 — APPROVED** [review]

File: Testing/functional/tests/cypress/integration/ui/errors/errorAlerts.js [hunk 2]
```diff
@@ -8,7 +8,8 @@ const tooManyRequestsStatusCode = 429;
 const serverErrorStatusCode = 500;
 
 function testGetConfigurationError(statusCode = serverErrorStatusCode) {
-    cy.enableModules([]);
+    cy.configureSettings({});
+    cy.wait(1000);
```
Comment (MikeLyttle) [inline Testing/functional/tests/cypress/integration/ui/errors/errorAlerts.js:12]: Can the cy.wait() be removed?

File: Testing/functional/tests/cypress/integration/ui/covid19/publicVaccineCard.js [hunk 3]
```diff
@@ -45,7 +45,16 @@ function clickVaccineCardEnterButton() {
...
+        cy.configureSettings(
+            {
+                covid19: {
+                    publicCovid19: {
+                        showFederalProofOfVaccination: true,
+                    },
+                },
+            },
+            ["Immunization"]
+        );
```
Comment (MikeLyttle) [inline Testing/functional/tests/cypress/integration/ui/covid19/publicVaccineCard.js:57]: I don't think showFederalProofOfVaccination needs to be set for this test.

File: Testing/functional/tests/cypress/integration/ui/errors/tooManyRequestsAlerts.js [hunk 4]
```diff
@@ -135,7 +153,18 @@ describe("Public Vaccine Card Downloads", () => {
...
+                covid19: {
+                    publicCovid19: {
+                        enableTestResults: true,
+                        showFederalProofOfVaccination: true,
+                    },
+                },
+            },
+            ["Immunization"]
+        );
+        //cy.enableModules(["Immunization", "PublicVaccineDownloadPdf"]);
```
Comment (MikeLyttle) [inline Testing/functional/tests/cypress/integration/ui/errors/tooManyRequestsAlerts.js:167]: I don't think enableTestResults needs to be set for this test.  Also, the commented out line can be removed.

**Review by ytqsl — APPROVED** [review]

File: Testing/functional/tests/cypress/integration/ui/pages/home.js [hunk 5]
```diff
@@ -18,6 +19,11 @@ describe("Authenticated User - Home Page", () => {
 
     it("Home - Federal Card button enabled", () => {
         cy.enableModules(["FederalCardButton"]);
+        cy.configureSettings({
+            homepage: {
+                showFederalProofOfVaccination: true,
+            },
+        });
```
Comment (MikeLyttle) [inline Testing/functional/tests/cypress/integration/ui/pages/home.js:26]: The enableModules call should be rolled into configureSettings.

File: Testing/functional/tests/cypress/integration/ui/pages/publicRoute.js [hunk 6]
```diff
@@ -11,6 +11,16 @@ describe("Public Route", () => {
...
+            {
+                covid19: {
+                    publicCovid19: {
+                        enableTestResults: true,
+                    },
+                },
+            },
+            ["Laboratory"]
+        );
         cy.enableModules(["Laboratory", "PublicLaboratoryResult"]);
```
Comment (MikeLyttle) [inline Testing/functional/tests/cypress/integration/ui/pages/publicRoute.js:24]: The enableModules call should be rolled into configureSettings.

File: Testing/functional/tests/cypress/integration/ui/user/waitlist.js [hunk 7]
```diff
@@ -30,6 +30,11 @@ function getTicket(status, queuePosition = 0) {
 describe("Waitlist Ticket Module Enabled", () => {
     beforeEach(() => {
         cy.logout();
+        cy.configureSettings({
+            waitingQueue: {
+                enabled: true,
+            },
+        });
         cy.enableModules(["Ticket"]);
```
Comment (MikeLyttle) [inline Testing/functional/tests/cypress/integration/ui/user/waitlist.js:38]: The enableModules call should be rolled into configureSettings.

File: Testing/functional/tests/cypress/integration/e2e/covid19/authenticatedVaccineCard.js (hunk 1 above)
Comment (BrianMaki) [reply Testing/functional/tests/cypress/integration/e2e/covid19/authenticatedVaccineCard.js:19]: No it has to be there.

File: Testing/functional/tests/cypress/integration/ui/errors/errorAlerts.js (hunk 2 above)
Comment (BrianMaki) [reply Testing/functional/tests/cypress/integration/ui/errors/errorAlerts.js:12]: Fixed in AB#14995

File: Testing/functional/tests/cypress/integration/ui/covid19/publicVaccineCard.js (hunk 3 above)
Comment (BrianMaki) [reply Testing/functional/tests/cypress/integration/ui/covid19/publicVaccineCard.js:57]: Fixed in Fixed in AB#14995

File: Testing/functional/tests/cypress/integration/ui/pages/home.js (hunk 5 above)
Comment (BrianMaki) [reply Testing/functional/tests/cypress/integration/ui/pages/home.js:26]: Fixed in Fixed in AB#14995

File: Testing/functional/tests/cypress/integration/ui/errors/tooManyRequestsAlerts.js (hunk 4 above)
Comment (BrianMaki) [reply Testing/functional/tests/cypress/integration/ui/errors/tooManyRequestsAlerts.js:167]: Fixed in AB#14995

File: Testing/functional/tests/cypress/integration/ui/user/waitlist.js (hunk 7 above)
Comment (BrianMaki) [reply Testing/functional/tests/cypress/integration/ui/user/waitlist.js:38]: Fixed in AB#14995

File: Testing/functional/tests/cypress/integration/ui/pages/publicRoute.js (hunk 6 above)
Comment (BrianMaki) [reply Testing/functional/tests/cypress/integration/ui/pages/publicRoute.js:24]: Fixed in AB#14995

**Pull request merged**